airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/JdbcDatabase.java

```diff
@@ -65,13 +66,15 @@ public void executeWithinTransaction(final List<String> queries) throws SQLExcep
    * @param <T> type that each record will be mapped to
    * @return stream of records that the result set is mapped to.
    */
-  public static <T> Stream<T> toStream(final ResultSet resultSet, final CheckedFunction<ResultSet, T, SQLException> mapper) {
+  @MustBeClosed
```

Comment:
this is cool! does intellij (and/or the compiler) actually show an error if it's not closed?

Reply:
No, IntelliJ does not do that.

By the way, this PR only deals with connection leaks for snowflake. The general problem will be handled in a separate PR.